pkg/interface/link/src/js/components/lib/link-detail-preview.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react';
+import { Route, Link } from "react-router-dom";
+import moment from "moment";
+
+export class LinkPreview extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      timeSinceLinkPost: this.getTimeSinceLinkPost(this.props.data)
+    };
+  }
+
+  componentDidUpdate(prevProps) {
```

Comment:
Can we order this to be below `componentDidMount`? Pretty sure all other components put them like that.